### PR TITLE
Enable CLI diagnostics by default and exit process when present

### LINKED ISSUE #1558: CLI builds should fail when there are errors present in the build
Add ability to break the build when there are error in the cli

### LINKED ISSUE #1595: Enable CLI diagnostics by default
`diagnostics` should be be `true` by default when executing `stc`.

can we do it with 4.x?

### PR DESCRIPTION
**Pull request by barak007**
fix(cli)!: report diagnostics true by default
* allow exit cleanly with ignoreDiagnostics option
BREAKING CHANGE:
* process will now exit with failure when there are any diagnostics

I'm not sure that this is the right setup between `diagnostics` and `ignoreDiagnostics` options. Please take a closer look.

closes #1595
closes #1558

### CODE REVIEW

**Comment by barak007** [comment]
fix: #1595 #1558

File: packages/cli/src/cli.ts [hunk 1]
```diff
@@ -192,6 +196,15 @@ build({
     useSourceNamespace: useNamespaceReference,
 });
 
+if (diagnosticsMessages.length) {
+    if (diagnostics) {
+        console.log('[Stylable Diagnostics]\n', diagnosticsMessages.join('\n\n'));
+    }
+    if (!ignoreDiagnostics) {
```
Comment (tomrav) [inline packages/cli/src/cli.ts:203]: Should we be more flexible in how we treat errors vs. warnings?

File: packages/cli/test/cli.spec.ts [hunk 2]
```diff
@@ -256,4 +256,35 @@ describe('Stylable Cli', () => {
         expect(stderr).equal('');
         expect(stdout).to.contain('I HAVE BEEN REQUIRED');
     });
+
+    describe('Cli diagnostics', () => {
```
Comment (tomrav) [inline packages/cli/test/cli.spec.ts:260]:
```suggestion
    describe('CLI diagnostics', () => {
```

**Review by tomrav — APPROVED** [review]
Looks good, strictness huzzah.

File: packages/cli/src/cli.ts (hunk 1 above)
Comment (barak007) [reply packages/cli/src/cli.ts]: yes.

**Pull request merged**